### DISCUSSION TITLE
Get type coerced data without form.handleSubmit?

### DISCUSSION THREAD

**Discussion by npearson72** [react-hook-form/react-hook-form#10069] [Q&A] [answered]
I'm using the Yup resolver with `useForm` and when I call `form.handleSubmit` I get a data object consisting of typed data based on my Yup schema.

I've noticed that `form.getValues` also provides the form data, but all the values are strings.

Is there a way to get the typed data that `form.handleSubmit` without submitting the form / using `form.handleSubmit`?

**Comment by leapful** [comment, marked as answer]
@npearson72 

HTML input element's value is string by default, it's not related to react-hook-form. If you want to get correct type value, please use option `valueAsNumber` or `valueAsDate` from `register` method, just keep note that you must set type on HTML input element too, like `<input type="number" />`.

Otherwise, you need to normalize (cast value to correct type) your form values manually  whenever you want to use it.
